treescript/tests/test_merges.py

```diff
@@ -188,3 +203,29 @@ def noop_replace(*arguments, **kwargs):
 
     assert len(called_args) == expected_calls
     assert result == expected_return
+
+
+@pytest.mark.asyncio
+async def test_bump_central(mocker, config, task, repo_context, merge_bump_info):
+    task["payload"]["merge_info"] = merge_bump_info
+    called_args = []
+
+    async def mocked_run_hg_command(config, *arguments, repo_path=None, **kwargs):
+        called_args.append([arguments])
+        if "return_output" in kwargs:
+            return "headers\n\n\n\n+invalid_changeset tag\n changeset tag\n+valid_changeset_is_forty_characters_long tag"
+
+    async def mocked_get_revision(*args, **kwargs):
+        return "some_revision"
+
+    async def noop_apply_rebranding(*arguments, **kwargs):
+        called_args.append("apply_rebranding")
+
+    mocker.patch.object(merges, "run_hg_command", new=mocked_run_hg_command)
+    mocker.patch.object(merges, "get_revision", new=mocked_get_revision)
+    mocker.patch.object(merges, "apply_rebranding", new=noop_apply_rebranding)
+
+    result = await merges.do_merge(config, task, repo_context.repo)
+
+    assert len(called_args) == 6
```

Comment:
Question: It's great to count the number of times `hg` was called. Do you think some calls are worth to be asserted? On top of my head, I think the tags and the commit ones are interesting and would be a good way to test their messages.

Reply:
Yes, I think you're probably right, I will add a few more detailed checks